scripts/pylib/twister/twisterlib.py

```diff
@@ -2310,8 +2310,9 @@ def run_build(self, args=[]):
             results = {'msg': msg, "returncode": p.returncode, "instance": self.instance}
 
             if out:
-                log_msg = out.decode(sys.getdefaultencoding())
-                with open(os.path.join(self.build_dir, self.log), "a") as log:
+                default_encoding = sys.getdefaultencoding()
```

Comment:
Could you please move this variable to some higher level, so there won't be a need to declare it several times in the following code? It is maybe a nitpick, but at least there will be less places where the code could potentially get out of sync